### DISCUSSION TITLE
Font rasterization algorithms

### DISCUSSION THREAD

**Discussion by page200** [mozilla/pdf.js#18701] [Q&A]
How do font rasterization algorithms in PDF readers in general and in PDF.js in particular manage the balancing act between their two conflicting goals:
- rendering text characters where they are located according to the PDF file,
- and ensuring nice [font hinting](https://en.wikipedia.org/wiki/Font_hinting)/sharpness/[kerning](https://en.wikipedia.org/wiki/Kerning)/legibility?

I mean low zoom levels where the pixel grid strongly affects the rendering result.

**Comment by calixteman** [comment]
About the PDF readers in general: I don't know.
About pdf.js: we just use the [canvas API](https://developer.mozilla.org/fr/docs/Web/API/Canvas_API) so the font rendering is up the web browsers (or whatever stuff which is implementing the canvas api).
